File: 후킹코드/level1/level1.js
setImmediate(function () { 
    /*setImmdediate 이벤트  주기 끝에 코드를 실행
    현재 이벤트 루프의 모든 입출력 작업 후 다음 이벤트 루프에 스케줄링된
    모든 타이머 이전에 실행된다. */
    Java.perform(function () { 
        var root_bypass=Java.use("java.lang.System"); //자바를 다루기 위한 Frida 함수
        root_bypass.exit.implementation = function(arg){ //System.exit(0)가 있는 메소드를 대체한다.
            console.log("root pass");
        }


        console.log("중간");

        /*
        //항상 트루로 우회
        var all_true=Java.use("sg.vantagepoint.uncrackable1.a");
        all_true.a.implementation= function(arg1){
            return true;
        }
        */

        //아스키 값 알아내기
        var decryptClass = Java.use("sg.vantagepoint.a.a");//이 클래스파일에서
		decryptClass.a.implementation = function(arg1, arg2){ //a메소드를 재구현 인자는 맞춰줘야함.
			var secret_string = this.a(arg1, arg2);
			console.log("\n[+] Secret String ASCII 값 :"+ secret_string + "\n");
			for(var i=0; i<secret_string.length; i++){
				console.log("[+] Secret String 값 :"+ String.fromCharCode(secret_string[i]));
			}
			return secret_string;
		}
    })



});

File: 후킹코드/level1/level1.py
import frida, sys

def on_message(message, data):
    print(message)

process_name = "uncrackable1"

#후킹 대상 함수가 존재하는 클래스 ,well-known 메서드는 공식문서 참고
#현재 클래스를 hook_target 으로 재구성한다.
#종료되는 걸 그냥 로그 찍는걸로 변경
jscode = """
console.log("[+] Start Script");

Java.perform(function(){
    console.log("[+] Hooking System.exit");
    var exitClass = Java.use("java.lang.System"); 
    exitClass.exit.implementation = function(){ 
        console.log("[+] System.exit called"); 
    }
});
"""

process = frida.get_usb_device(1).attach(process_name)#장치에서 패키지에 맞는 프로세스 가져옴
script = process.create_script(jscode)
script.on("message", on_message)
print("[+] Running Hook")
script.load()
sys.stdin.read()


